src/Games/NexusMods.Games.AdvancedInstaller.UI/AdvancedManualInstallerUI.cs

```diff
@@ -0,0 +1,126 @@
+            loadout?.Mods.TryGetValue(baseModId, out mod);
+        }
+
+        var modName = mod?.Name ?? Language.AdvancedInstaller_Manual_Mod;
+
+
+        // Note: This code is effectively a stub.
+        var (shouldInstall, deploymentData) = await GetDeploymentDataAsync(gameInstallation, modName,
+            archiveFiles);
+
+        if (!shouldInstall)
+            return Array.Empty<ModInstallerResult>();
+
+        return new[]
+        {
+            new ModInstallerResult
+            {
+                Id = baseModId,
+                Files = deploymentData.EmitOperations(archiveFiles)
+            }
+        };
+    }
+
+
+    private async Task<(bool shouldInstall, DeploymentData data)> GetDeploymentDataAsync(
+        GameInstallation gameInstallation, string modName,
+        FileTreeNode<RelativePath, ModSourceFileEntry> archiveFiles)
+    {
+        var installerViewModel =
+            new AdvancedInstallerWindowViewModel(modName, archiveFiles, gameInstallation.LocationsRegister,
+                gameInstallation.Game.Name);
+
+        await ShowAdvancedInstallerDialog(installerViewModel);
+
+        return (installerViewModel.AdvancedInstallerVM.ShouldInstall,
+            installerViewModel.AdvancedInstallerVM.BodyViewModel.DeploymentData);
+    }
+
+    /// <summary>
+    /// Creates a modal Dialog window and shows it, then awaits for it to close.
+    /// </summary>
+    /// <param name="dialogVM">The View Model of the dialog to create.</param>
+    private static async Task ShowAdvancedInstallerDialog(IAdvancedInstallerWindowViewModel dialogVM)
+    {
+        var tcs = new TaskCompletionSource();
+
+        OnUi((dialogVM, tcs), async tuple =>
+        {
+            var view = new AdvancedInstallerWindowView
+            {
+                ViewModel = tuple.dialogVM
+            };
+
+            // Get the main window.
+            if (Application.Current?.ApplicationLifetime is
+                IClassicDesktopStyleApplicationLifetime { MainWindow: not null } desktop)
+            {
+                // Create the modal dialog, parent window is required for modal.
+                await view.ShowDialog(desktop.MainWindow);
+            }
+
+            tcs.SetResult();
+        });
+
+        await tcs.Task;
+    }
+
+    /// <summary>
+    /// Schedule an action on the UI thread.
+    /// This returns immediately, without waiting for the action to actually run.
+    /// </summary>
+    /// <param name="state">State data required for the execution of the action.</param>
+    /// <param name="action">The action to execute.</param>
+    /// <typeparam name="TState">The type of the state data.</typeparam>
+    private static void OnUi<TState>(TState state, Func<TState, Task> action)
+    {
+        AvaloniaScheduler.Instance.ScheduleAsync(state: (state, action),
+            async (_, innerState, _) => { await innerState.action(innerState.state); });
+    }
+}
```

Comment:
I feel like this is the 10th custom `OnUi` method. There are a couple of these in the FOMOD UI project as well. We should just consolidate all of them in the `NexusMods.App.UI` project.